### PR TITLE
Fix "no-did-mount-set-state" eslint error in tile-validator.js

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**
Replace `componentDidMount` with `getDerivedStateFromProps`.

### CODE REVIEW

File: examples/experimental/i3s-17-and-debug/components/tile-validator.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ const propTypes = {
 };
 
 const defaultProps = {
-  tile: null,
+  tile: {},
```
Comment (ibgreen) [inline examples/experimental/i3s-17-and-debug/components/tile-validator.js:79]: This seems wrong. I think a value should either be a valid tile or null. Allowing a value to take on an "invalid value" seems to do more harm than good.

Isn't there a `PropTypes.optional.object`.

File: examples/experimental/i3s-17-and-debug/components/tile-validator.js [hunk 2]
```diff
@@ -88,18 +88,23 @@ export default class TileValidator extends PureComponent {
...
     };
   }
 
-  componentDidUpdate(prevProps) {
-    const isNoTile = !this.props.tile || !prevProps.tile;
-
-    if (isNoTile || this.props.tile.id !== prevProps.tile.id) {
-      this.setState({geometryInfo: null, triangleMessages: null, boundingVolumeInfo: null});
+  static getDerivedStateFromProps(props, state) {
+    if (props.tile.id !== state.tileId) {
```
Comment (ibgreen) [inline examples/experimental/i3s-17-and-debug/components/tile-validator.js:99]: Maybe a line of comment about what is happening here?

**Review by ibgreen — APPROVED** [review]

File: examples/experimental/i3s-17-and-debug/components/tile-validator.js (hunk 2 above)
Comment (ibgreen) [reply examples/experimental/i3s-17-and-debug/components/tile-validator.js:99]: Something like `// New tile is being set, re-initialize state`

File: examples/experimental/i3s-17-and-debug/components/tile-validator.js (hunk 1 above)
Comment (dryabinin-actionengine) [reply examples/experimental/i3s-17-and-debug/components/tile-validator.js:79]: returned back

File: examples/experimental/i3s-17-and-debug/components/tile-validator.js (hunk 2 above)
Comment (dryabinin-actionengine) [reply examples/experimental/i3s-17-and-debug/components/tile-validator.js:100]: Added

**Pull request merged**